pkg/proc/amd64_arch.go

```diff
@@ -219,7 +219,7 @@ func amd64SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 		return true
 
 	default:
-		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.fatalthrow" {
+		if it.systemstack && it.top && it.g != nil && strings.HasPrefix(it.frame.Current.Fn.Name, "runtime.") && it.frame.Current.Fn.Name != "runtime.throw" {
```

Comment:
I think this shouldn't change. Just the place where we set the breakpoint.

Reply:
Without this change `TestCgoStacktraces2` will fail, we still need to prevent switching to the goroutine stack, and since we will no longer see `runtime.fatalthrow` in the stacktrace we have to check for `runtime.throw` instead. I could keep `runtime.fatalthrow` and just add this as an OR condition.

Reply:
Yes, I think adding an OR is the right solution. Even though we don't stop on runtime.fatalthrow deliberately anymore the program could end up stopped there anyway.